tests/clv/models/test_beta_geo.py

```diff
@@ -207,20 +253,41 @@ def test_expected_probability_alive(self):
             rtol=0.05,
         )
 
+    def test_fit_result_without_fit(self, data, model_config):
+        model = BetaGeoModel(data=data, model_config=model_config)
+        with pytest.raises(RuntimeError, match="The model hasn't been fit yet"):
+            model.fit_result
+
+        idata = model.fit(
+            tune=5,
+            chains=2,
+            draws=10,
+            compute_convergence_checks=False,
+        )
+        assert isinstance(idata, az.InferenceData)
+        assert len(idata.posterior.chain) == 2
+        assert len(idata.posterior.draw) == 10
+        assert model.idata is idata
+
     def test_expected_num_purchases(self):
         customer_id = np.arange(10)
         test_t = np.linspace(20, 38, 10)
         test_frequency = np.tile([1, 3, 5, 7, 9], 2)
         test_recency = np.tile([20, 30], 5)
         test_T = np.tile([25, 35], 5)
-
-        bg_model = BetaGeoModel(
-            customer_id=customer_id,
-            frequency=test_frequency,
-            recency=test_recency,
-            T=test_T,
+        data = pd.DataFrame(
+            {
+                "customer_id": customer_id,
+                "frequency": test_frequency,
+                "recency": test_recency,
+                "T": test_T,
+            }
         )
-        bg_model._fit_result = az.from_dict(
+
+        bg_model = BetaGeoModel(data=data)
+        bg_model.build_model()
+        bg_model.fit("map")
```

Comment:
Why is this needed?

Reply:
fit_result property takes value from class.idata['posterior'], before the fit() is performed class.idata doesn't exist, so trying to overwrite it would fail

Reply:
So can we set class.idata manually?

Reply:
no longer using fit in this test